### PR TITLE
Fix assignment of references

### LINKED ISSUE #234: Setting a PHP::GLOBALS value makes a segfault
Hello, trying to set a default global value using a static method cause a segfault.
Inspecting the code, seems that on Global.cpp -> Global &Global::update() the active_symbol_table isn't already built.
Adding this check before the zend_hash_add solves the segfault, but when i try to read this value with PHP::GLOBALS it's always null (maybe because Globals::operator[] read from symbol_table and not from active_symbol_table ???).

```
if(!EG(active_symbol_table))
  zend_rebuild_symbol_table(TSRMLS_C);
```

I'm missing something?
Thank you.

### LINKED ISSUE #269: Process termination when attempting to access object
After spending a few hours debugging through native code that worked on PHP 5.6 (and PHP-CPP legacy), it looks like the recent 2.0.0 release is not playing nice when objects are passed in.

I've managed to whittle down the problem to what essentially looks like an issue with accessing PHP objects.

Example:

``` c
#include <phpcpp.h>
#include <iostream>
void test(Php::Parameters &params)
{       
    Php::Value obj = params[0]; 
    std::cout << obj["test"] << std::endl;
}

extern "C" {
    PHPCPP_EXPORT void *get_module() {
        static Php::Extension extension("php_cpp", "1.0");
        extension.add<test>("test", { 
            Php::ByRef("obj",` Php::Type::Null, false);
        });
        return extension;
    }
}
```

And then a test script to confirm:

``` php
$foo = new stdClass;
$foo->test = 23;
test($foo);
```

Results in a segfault.  Passing `$foo = ['test' => 23]` (as an array) to the function has no issues.

Tested on PHP-CLI 7.0.12. The header file being used reported by php-config also matches this version, so I do not believe there were any compilation issues when building 2.0.

### PR DESCRIPTION
**Pull request by sjinks**
Fixes #269 and allows to fix #234

### CODE REVIEW

File: zend/value.cpp [hunk 1]
```diff
@@ -203,12 +203,12 @@ Value::Value(const IniValue &value) : Value((const char *)value) {}
...
 {
-    if (Z_ISREF_P(that._val)) {
-        zend_assign_to_variable(_val, that._val, IS_VAR);
-    }
-    else {
-        ZVAL_COPY(_val, that._val);
+    zval* z = that._val;
+    while (Z_ISREF_P(z)) {
+        ZVAL_DEREF(z);
     }
```
Comment (sjinks) [inline zend/value.cpp]: When copying a variable which is a reference to another variable, we need to get to the original variable and copy it, NOT create a copy of the reference itself. This differs from PHP5, as PHP7 has a dedicated IS_REFERENCE type for references.

File: zend/value.cpp [hunk 2]
```diff
@@ -335,7 +335,14 @@ Value &Value::operator=(const Value &value)
     // skip self assignment
     if (this == &value) return *this;
 
-    zend_assign_to_variable(_val, value._val, IS_VAR);
+    if (Z_REFCOUNTED_P(_val) || Z_ISREF_P(_val) || Z_ISREF_P(value._val)) {
+        zend_assign_to_variable(_val, value._val, IS_CV);
```
Comment (sjinks) [inline zend/value.cpp]: I plan to get rid eventually of zend_assign_to_variable() but need to smoke something funny to understand how it works — chances are that `IS_XXX` flag will depend on the type of the variable. Though right now [all my tests](https://github.com/sjinks/PHP-CPP/tree/tests/test) pass, I don't feel confident about it.

**Comment by sjinks** [comment]
Assignments should work fine now.

**Comment by martijnotto** [comment]
Thanks! Could you please put in some comments describing the new code?

**Comment by sjinks** [comment]
OK.

**Comment by sjinks** [comment]
Done. If there are unclear moments, let us review them.

**Comment by martijnotto** [comment]
Thanks. I have merged your PR. I made the makeReference function non-static as I didn't really see a reason for it to be static in the first place.

I also renamed some variables and added some more comments for clarity. I also moved curly braces to their own line for consistency.

**Comment by sjinks** [comment]
> I made the makeReference function non-static as I didn't really see a reason for it to be static in the first place

Well, it was supposed to be used like this:

```c++
Php::Value variable = 1;
Php::Value reference = Php::Value::makeReference(variable);
```

That is, it acts just like a constructor, it creates a new Value which is set as a reference to another Value.

**Comment by sjinks** [comment]
I must say your implementation of `makeReference()` looks better.

**Pull request merged**